device/samples/pnp/pnpTemperatureController.js

```diff
@@ -0,0 +1,256 @@
+  helperLogCommandRequest(request);
+  switch (request.methodName) {
+  case commandNameGetMaxMinReport1: {
+    await sendCommandResponse(request, response, 200, 'max min report from thermostat 1');
+    break;
+  }
+  case commandNameGetMaxMinReport2: {
+    await sendCommandResponse(request, response, 200, 'max min report from thermostat 2');
+    break;
+  }
+  case commandNameReboot: {
+    await sendCommandResponse(request, response, 200, 'reboot response');
+    break;
+  }
+  default:
+    await sendCommandResponse(request, response, 404, 'unknown method');
+    break;
+  }
+};
+
+const sendCommandResponse = async (request, response, status, payload) => {
+  try {
+    await response.send(status, payload);
+    console.log('Response to method \'' + request.methodName +
+              '\' sent successfully.' );
+  } catch (err) {
+    console.error('An error ocurred when sending a method response:\n' +
+              err.toString());
+  }
+};
+
+const helperLogCommandRequest = (request) => {
+  console.log('Received command request for comand name \'' + request.methodName + '\'');
+
+  if (!!(request.payload)) {
+    console.log('The command request paylaod :');
+    console.log(request.payload);
+  }
+};
+
+
+const helperCreateReportedPropertiesPatch = (propertiesToReport, componentName) => {
+    let patch;
+    if (!!(componentName)) {
+      patch = { };
+      propertiesToReport.__t = "c";
+      patch[componentName] = propertiesToReport;
+    }
+    else {
+      patch = { };
+      patch = propertiesToReport;
+      }
+    if (!!(componentName)) {
+    console.log('The following properties will be updated for component:' + componentName);
+    }
+    else{
+      console.log('The following properties will be updated for root interface:');
+    }
+    console.log(patch);
+    return patch;
+  };
+
+const updateComponentReportedProperties = (deviceTwin, patch, componentName) => {
+  let logLine;
+  if (!!(componentName))
+  {
+    logLine = 'Properties have been reported for component:' + componentName;
+  }
+  else{
+    logLine = 'Properties have been reported for root interface';
+  }
+  deviceTwin.properties.reported.update(patch, function (err) {
+    if (err) throw err;
+    console.log(logLine);
+  });
+};
+
+const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin) => {
+
+  deviceTwin.on('properties.desired', (delta) => {
+    if (delta[thermostat1ComponentName] || delta[thermostat2ComponentName] || delta[deviceInfoComponentName]) {
+    const version = delta.$version;
+    const componentName = Object.keys(delta)[0];
+    let logLine;
+    if (!!(componentName))
+    {
+      logLine = 'Received an update for component: ' + componentName + ' with value: ' + JSON.stringify(delta);
+    }
+    else{
+      logLine = 'Received an update for root interface with value: ' + JSON.stringify(delta);
+    }
+    console.log(logLine);
+    const patch = { [componentName]: {} };
```

Comment:
>[componentName] [](start = 20, length = 15)

This works out ok for the no componentName case?  patch will be undefined.

Reply:
code has been changed now...but currently for this method no component will not work. I have a TODO here